lib/internal/buffer.js

```diff
@@ -1,4 +1,13 @@
 'use strict';
 
-// This is needed still for FastBuffer
-module.exports = {};
+const binding = process.binding('buffer');
+const { setupBufferJS } = binding;
+
+// Remove from the binding so that function is only available as exported here.
+// (That is, for internal use only.)
+delete binding.setupBufferJS;
```

Comment:
Not blocking, and up for debate:
Since this changes global state, maybe move this to `node_bootstap`, or explicitly require with a comment. Otherwise AFAICT it will run as a side effect of https://github.com/nodejs/node/blob/02a52670b832b17b90365aa42b5620c20fe357f6/lib/internal/bootstrap_node.js#L295 which is not obvius

Reply:
Hmm, I'd like to keep the caching and removing of `setupBufferJS` in the same place. Would a comment in `bootstrap_node.js` clarifying that requiring `buffer` also removes the `setupBufferJS` from the binding be sufficient to clarify? Something like:

```js
// Note that this requires `lib/internal/buffer.js`, which removes `setupBufferJS`
// from the buffer binding.
global.Buffer = NativeModule.require('buffer').Buffer;
```

Reply:
Another viable option is to add this to `bootstrap_node.js`
```js
// This require as side effect removes `setupBufferJS` from the buffer binding.
NativeModule.require('internal/buffer.js');
```

Reply:
Cool, I'll throw a comment in. Stay tuned.